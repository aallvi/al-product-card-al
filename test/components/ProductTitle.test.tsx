// import * as React from 'react';
import React from "react";
import renderer from 'react-test-renderer'
import ProductCard, {  ProductTitle } from '../../src/components'
import { product,  } from "../data/products";




describe('ProductTitle', () => { 


    test('debe mostrar el componente correctamente con el titulo personalizado', () => { 

            const wrapper = renderer.create(
                <ProductTitle title="Custom Product" className="custom-class" />
            )

         
            expect (wrapper.toJSON()).toMatchSnapshot()

     })


     test('should show component with name of the product', () => { 

        const wrapper = renderer.create(
                    <ProductCard  product={product} >

                {     () => (
                            <ProductTitle/>
                        )
                    }

                    </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot()




      })



 })