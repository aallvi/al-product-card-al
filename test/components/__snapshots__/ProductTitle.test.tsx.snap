// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle debe mostrar el componente correctamente con el titulo personalizado 1`] = `
<span
  className="custom-class productDescription "
>
   
  Custom Product
   
</span>
`;

exports[`ProductTitle should show component with name of the product 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="undefined productDescription "
  >
     
    Coffe mug - SIN IMAGEN
     
  </span>
</div>
`;
